src/coreclr/src/tools/Common/Compiler/VectorFieldLayoutAlgorithm.cs

```diff
@@ -90,7 +90,8 @@ public override bool ComputeContainsGCPointers(DefType type)
 
         public override ValueTypeShapeCharacteristics ComputeValueTypeShapeCharacteristics(DefType type)
         {
-            if (type.Context.Target.Architecture == TargetArchitecture.ARM64)
+            if (type.Context.Target.Architecture == TargetArchitecture.ARM64 &&
+                type.Instantiation[0].IsPrimitiveNumeric)
```

Comment:
Should this be a special `IsSupportedVectorBaseType` instead? We only support the 10 primitive types today, but that may change in the future when we add `System.Half` for example (which is a supported vector type but which isn't an IL primitive).

Reply:
That follows naming of the API, `getTypeForPrimitiveNumericClass` is used to recognize those 10 allowed primitive types here: https://github.com/dotnet/runtime/blob/e5badd7a351a04b8b14cb5ec2c7e1e8853991ad7/src/coreclr/src/jit/simd.cpp#L580
If we add more allowed types, we should rename the API and this property at the same time.

Reply:
I assume that the fact we don't support this with `nint` and `nuint` (IntPtr/UIntPtr) is intentional.

Reply:
It is currently intentional but there is a proposal to extend support to those types as well: https://github.com/dotnet/runtime/issues/36160

Reply:
Then we might want to throw a `RequiresRuntimeJitException` for `Vector*<nint>` types to avoid a breaking change (passing in X vs. Q registers) later.